vignettes/update.Rmd

```diff
@@ -0,0 +1,254 @@
+
+```{r}
+wf3 <- wf2 %>% remove_frosting()
+p3 <- predict(wf3, latest)
+p3
+```
+You can see that the above results from `p3` are the same as from `p1`, when we simply have a prediction layer in the `frosting` post-processing container.
+
+## Adjust a single step of an `epi_recipe`
+
+Suppose that we just want to change a single step in an `epi_recipe` (that is either standalone or a part of an `epi_workflow`). Instead of replacing an entire `epi_recipe`, we can use the `adjust_epi_recipe` function. In this function, the step to be adjusted is indicated either the step number or name in the `which_step` parameter. Then, the parameter name and update value must be inputted as `...`.
+
+For instance, suppose that we decide to lead the `death_rate` by 14 days instead of 7. We may adjust this step in `wf` recipe by setting `which_step` to the step number in the order of operations, which can be obtained by inspecting `r2` or the tidy summary of it:
+
+```{r}
+workflows::extract_preprocessor(wf) # step_epi_ahead is the third step in r2
+tidy(workflows::extract_preprocessor(wf)) # tidy tibble summary of r2
+
+wf <- wf %>% adjust_epi_recipe(which_step = 3, ahead = 14)
+```
+
+Alternatively, we may adjust that step by name by specifying the full name of the step, `step_epi_ahead`, in `which_step`:
+
+```{r}
+wf %>% adjust_epi_recipe(which_step = "step_epi_ahead", ahead = 14) # not overwrite r2 because same result
+```
+
+If there are at least two steps in a recipe that share the same name, specifying the name in `which_step` will throw an error as `adjust_epi_recipe` is not intended to be used to modify multiple steps at once. The way, then, to modify a step that has the same name as another is to indicate what number it is in the ordering of the steps. For example, in `r2` there are two steps named `step_epi_lag` - the first step where we lag the death rate, and the second where we lag the case rate. If we want to modify the lags for the `case_rate` variable, we would specify the step number of 2 in `which_step`. 
+
+```{r}
+wf <- wf %>% adjust_epi_recipe(which_step = 2, lag = c(0, 1, 7, 14, 21))
+
+workflows::extract_preprocessor(wf)
+```
+
+We could adjust a recipe directly in the same way as we adjust a recipe in a workflow. The main difference is that we would not input `wf` as the first argument to `adjust_epi_recipe` but rather `r2`.
+
+```{r}
+adjust_epi_recipe(r2, which_step = 2, lag = c(0, 1, 7, 14, 21)) # should be same result as above
+```
+
+Note that when we adjust the `r2` object directly, we are not adjusting the recipe in the `epi_workflow`. That is, if we modify a step in `r2`, the change will not automatically transfer over to `wf`. We would need to modify the recipe in `wf` directly (`adjust_epi_recipe` on `wf`) or update the recipe in `wf` with a new `epi_recipe` that has undergone the adjustment (using `update_epi_recipe`):
+
+```{r}
+r2 <- adjust_epi_recipe(r2, which_step = 2, lag = 0:21)
+
+workflows::extract_preprocessor(wf)
```

Comment:
```suggestion
extract_preprocessor(wf)
```

Reply:
I think I'm going to keep the `workflows::` for now because it's a quick and clear way to indicate what functions I used are from that package & I think it's good to lay that out somewhere (because otherwise it can be hard to tell for a beginner which functions are from that package and which are from ours - because it's not exceedingly common).